### PR TITLE
✨(tray) move to k8s

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

Following recent work on migrating our apps to k8s [1], ralph was missing to the list!

## Proposal

- [x] move tray templates to k8s objects
- [x] update the CI to use `kubectl` and `k3d` instead of `oc`

References:

[1] https://github.com/openfun/arnold/pull/620

### CODE REVIEW

**Comment by jmaupetit** [comment]
Blocked by https://github.com/openfun/arnold/pull/660

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +26,7 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 ### Fixed
 
 - Tray: remove `version` immutable field in DC selector
+- Tray: remove OKD compatibility
```
Comment (jmaupetit) [inline CHANGELOG.md:29]: Should be in the _removed_ section?

Comment (quitterie-lcs) [reply CHANGELOG.md:29]: Yes and both of them, no?

Comment (madmatah) [reply CHANGELOG.md:29]: I'd merge the 2 lines into one and put it in the `Changed` section :sweat_smile:

File: src/tray/templates/services/app/deploy.yml.j2 [hunk 2]
```diff
@@ -43,10 +48,13 @@ spec:
             exec:
               command: ["ralph", "--help"]
             initialDelaySeconds: 3
-            periodSeconds: 1
+            # ralph can be slow to boot
```
Comment (madmatah) [inline src/tray/templates/services/app/deploy.yml.j2:51]: This comment is related to `initialDelaySeconds`, isn't it ?

File: src/tray/templates/services/app/deploy.yml.j2 [hunk 3]
```diff
@@ -43,10 +48,13 @@ spec:
...
+            timeoutSeconds: 10
           readinessProbe:
             exec:
               command: ["ralph", "--help"]
+            # ralph can be slow to boot
```
Comment (madmatah) [inline src/tray/templates/services/app/deploy.yml.j2:56]: Ditto, the solution for a process slow to boot is `initialDelaySeconds` IMO.

File: .circleci/config.yml [hunk 4]
```diff
@@ -216,40 +216,69 @@ jobs:
...
+
+      - run:
+          name: Configure sysctl limits for Elasticsearch
           command: |
             # Elasticsearch requires to increase this setting's default value
             sudo sysctl -w vm/max_map_count=262144
+
+      - run:
+          name: Configure Docker for k3d cluster
+          command: |
```
[the remark this answers is not in the document]
Comment (jmaupetit) [reply .circleci/config.yml:243]: True.

File: src/tray/templates/services/app/deploy.yml.j2 (hunk 2 above)
Comment (madmatah) [reply src/tray/templates/services/app/deploy.yml.j2:51]: Oops, forget what I said :sweat_smile:

File: src/tray/templates/services/app/deploy.yml.j2 (hunk 3 above)
Comment (madmatah) [reply src/tray/templates/services/app/deploy.yml.j2:56]: Oops, forget what I said :sweat_smile:

**Comment by jmaupetit** [comment]
I've addressed all your comments @madmatah && @quitterie-lcs // Ready for a final review! :pray:

**Review by madmatah — APPROVED** [review]

**Pull request merged**